doc/man7/provider-mac.pod

```diff
@@ -0,0 +1,249 @@
+ OP_mac_ctx_set_params       OSSL_FUNC_MAC_CTX_SET_PARAMS
+
+ OP_mac_get_param_types      OSSL_FUNC_MAC_GET_PARAM_TYPES
+ OP_mac_ctx_get_param_types  OSSL_FUNC_MAC_CTX_GET_PARAM_TYPES
+ OP_mac_ctx_set_param_types  OSSL_FUNC_MAC_CTX_SET_PARAM_TYPES
+
+A mac algorithm implementation may not implement all of these functions.
+In order to be a consistent set of functions, at least the following functions
+must be implemented: OP_mac_newctx(), OP_mac_freectx(), OP_mac_init(),
+OP_mac_update(), OP_mac_final().
+All other functions are optional.
+
+=head2 Context Management Functions
+
+OP_mac_newctx() should create and return a pointer to a provider side
+structure for holding context information during a mac operation.
+A pointer to this context will be passed back in a number of the other mac
+operation function calls.
+The paramater B<provctx> is the provider context generated during provider
+initialisation (see L<provider(3)>).
+
+OP_mac_freectx() is passed a pointer to the provider side mac context in
+the B<mctx> parameter.
+This function should free any resources associated with that context.
+
+OP_mac_dupctx() should duplicate the provider side mac context in the
+B<mctx> parameter and return the duplicate copy.
+
+=head2 Encryption/Decryption Functions
+
+OP_mac_init() initialises a mac operation given a newly created provider
+side mac context in the B<mctx> paramter.
+
+OP_mac_update() is called to supply data for MAC computation of a previously
+initialised mac operation.
+The B<mctx> parameter contains a pointer to a previously initialised provider
+side context.
+OP_mac_update() may be called multiple times for a single mac operation.
+
+OP_mac_final() completes the MAC computation started through previous
+OP_mac_init() and OP_mac_update() calls.
+The B<mctx> parameter contains a pointer to the provider side context.
+The resulting MACshould be written to B<out> and the amount of data written
+to B<*outl>, which should not exceed B<outsize> bytes.
+The same expectations apply to B<outsize> as documented for
+L<EVP_MAC_final(3)>.
+
+=head2 Mac Parameters
+
+See L<OSSL_PARAM(3)> for further details on the parameters structure used by
+these functions.
+
+OP_mac_get_params() gets details of parameter values associated with the
+provider algorithm and stores them in B<params>.
+
+OP_mac_ctx_set_params() sets mac parameters associated with the given
+provider side mac context B<mctx> to B<params>.
+Any parameter settings are additional to any that were previously set.
+
+OP_mac_ctx_get_params() gets details of currently set parameter values
+associated with the given provider side mac context B<mctx> and stores them
+in B<params>.
+
+OP_mac_get_param_types(), OP_mac_ctx_get_param_types(), and
+OP_mac_ctx_set_param_types() all return constant B<OSSL_PARAM> arrays
+as descriptors of the parameters that OSSL_mac_get_params(),
+OSSL_mac_ctx_get_params(), and OSSL_mac_ctx_set_params() can handle,
```

Comment:
Do you mean OP_mac_ctx_set_params() ? This is the only mention of this name in thhis pod file??

Reply:
Oops!  Yes, I do.  Darn slippery fingers :wink: